### PR TITLE
Consume upjet ProviderScheduler

### PR DESCRIPTION
**Pull request by ulucinar**
### Description of your changes

Associated with: #325

This PR reenables the shared gRPC server runtime previously disabled due to external resource leakage issues. It consumes the new provider scheduler from upjet and:
- Enables the shared server runtime
- Adds `--provider-ttl` command-line option through which the max TTL used by the shared scheduler can be configured.

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

This PR has been tested in the experiments done here:
https://github.com/upbound/provider-aws/issues/576

### CODE REVIEW

**Comment by ulucinar** [comment]
/test-examples="examples/ec2/vpc.yaml"

File: internal/clients/provider_config.go [hunk 1]
```diff
@@ -51,6 +51,7 @@ const (
 	errAWSConfigUpbound     = "failed to get AWS config using Upbound identity"
 
 	upboundProviderIdentityTokenFile = "/var/run/secrets/upbound.io/provider/token"
+	defaultIdentityTokenFile         = "/var/run/secrets/eks.amazonaws.com/serviceaccount/token"
```
Comment (sergenyalcin) [inline internal/clients/provider_config.go:54]: It seems that we do not use this constant. Could you please remove it?

**Review by sergenyalcin — COMMENTED** [review]
Thanks @ulucinar I left a comment.

**Comment by sergenyalcin** [comment]
Some test results that are done via this branch content:

https://github.com/upbound/provider-aws/issues/576#issuecomment-1485259533
https://github.com/upbound/provider-aws/issues/576#issuecomment-1474137489
https://github.com/upbound/provider-aws/issues/576#issuecomment-1482747981

**Comment by ulucinar** [comment]
/test-examples="examples/ec2/vpc.yaml"

**Comment by sergenyalcin** [comment]
/test-examples="examples/ec2/vpc.yaml"

File: internal/clients/provider_config.go (hunk 1 above)
Comment (ulucinar) [reply internal/clients/provider_config.go]: Done. Thanks for catching it.

**Comment by ulucinar** [comment]
`make lint` runs successfully locally, merging the PR:
image

**Pull request merged**